### PR TITLE
Add namespace MetaGen to the generic Resource

### PR DESCRIPTION
**Pull request by tetianakravchenko**
Why: If kubernetes resource is namespaced, we want to add namespace labels/annotations to the corresponding resource, according to the `add_resource_metadata` configuration. Related issue: https://github.com/elastic/beats/issues/33108

Later I am planning to use `NewNamespaceAwareResourceMetadataGenerator`, instead of `NewResourceMetadataGenerator` [here](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/util/kubernetes.go#L108) 

`namespaceMeta` is already created for service resource -
https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/util/kubernetes.go#L111

Related to: https://github.com/elastic/beats/issues/33144

### CODE REVIEW

**Comment by MichaelKatsoulis** [comment]
@tetianakravchenko with this PR you are not adding any new functionality, right? You just crate the `NewNamespaceAwareResourceMetadataGenerator` so that pod and service `GenerateK8s` do not need to duplicate the same functionality for adding namespace meta. And then when merged you can use this new function in the modules.

**Comment by tetianakravchenko** [comment]
> @tetianakravchenko with this PR you are not adding any new functionality, right? You just crate the NewNamespaceAwareResourceMetadataGenerator so that pod and service GenerateK8s do not need to duplicate the same functionality for adding namespace meta. And then when merged you can use this new function in the modules.

@MichaelKatsoulis yes, I've just moved adding namespace meta to the generic Resource, so `pod`, `service` and other Resources defined [here](https://github.com/elastic/beats/blob/main/metricbeat/module/kubernetes/util/kubernetes.go#L119-L164) can use it - in this way there is no need to define which resource is namespaced, and which is not - it will be defined based on `accessor.GetNamespace()` - [here](https://github.com/elastic/elastic-agent-autodiscover/pull/33/files#diff-753cb073cbf0e1ffa3be26aa8f16a1e83e4d8da767a6a0b65bc0d4d936a78130R126-R127)

**Review by MichaelKatsoulis — APPROVED** [review]

File: kubernetes/metadata/resource.go [hunk 1]
```diff
@@ -113,8 +123,18 @@ func (r *Resource) GenerateK8s(kind string, obj kubernetes.Resource, options ...
...
-		_ = safemapstr.Put(meta, "namespace", accessor.GetNamespace())
+	namespaceName := accessor.GetNamespace()
+	if namespaceName != "" {
+		_ = safemapstr.Put(meta, "namespace", namespaceName)
+
+		if r.namespace != nil {
+			nsMeta := r.namespace.GenerateFromName(namespaceName)
+			if nsMeta != nil {
+				// Use this in 8.0
+				//out.Put("namespace", meta["namespace"])
```
Comment (ChrsMark) [inline kubernetes/metadata/resource.go:134]: Most probably this is not needed. We have concluded that we are not changing anything here. You can remove  completely  this comment :)

**Review by ChrsMark — APPROVED** [review]
lgtm!

**Pull request merged**